src/convertToObject.js

```diff
@@ -3,16 +3,35 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const propertiesArray = sourceString.split(';');
+  let cssProperties = {};
+
+  for (const property of propertiesArray) {
+    const pairOfInternals = property.split(':');
+
+    if (pairOfInternals[0] && pairOfInternals[1]) {
+      const prop = pairOfInternals[0].trim();
+      const value = pairOfInternals[1].trim();
+      const objOfInternals = {
+        [prop]: value,
+      };
+
+      cssProperties = Object.assign(cssProperties, objOfInternals);
```

Comment:
agree with @mitattt . You should use reduce to create a new object. I understand, it's brain damage, because i was in the same situation as you  😭